Documentation/Makefile

```diff
@@ -1,93 +1,44 @@
-	$(QUIET) ${AGENTDIR}/cilium-agent --cmdref $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-health
-	$(QUIET) ${HEALTHDIR}/cilium-health --cmdref $(CMDREFDIR)
-	@$(ECHO_GEN)cmdref/cilium-operator
-	$(QUIET) ${OPERATORDIR}/cilium-operator --cmdref $(CMDREFDIR)
+clean:
+	-$(QUIET)rm -rf _build
 
-# Touch "$@.ok" if the docs build succeeds. Fail if there are errors
-# or output other than those including the following phrases:
-# "tabs assets" - These are always printed by the spelling check.
-# "misspelled words" - These will be explicitly checked below.
-# Finally, fail out if we can't remove the file that marks success.
-check-build: check-requirements
-	@$(ECHO_CHECK) documentation build...
-	@mkdir -p $(BUILDDIR)
-	$(QUIET)($(SPHINXBUILD) -b spelling -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)/spelling" -q -E $(SPHINXOPTS) 2>&1 && touch $@.ok) > $(BUILDDIR)/warnings.txt
-	@if grep -q -v $(IGNOREDWARNINGS) $(BUILDDIR)/warnings.txt; then \
-		echo "Please fix the following documentation warnings:"; \
-		grep -v $(IGNOREDWARNINGS) "$(BUILDDIR)/warnings.txt"; \
-		exit 1; \
-	fi
-	@if [ "$$(wc -l < "$(BUILDDIR)/spelling/output.txt")" -ne "0" ]; then \
-	        echo ""; \
-		echo "Please fix the following spelling mistakes:"; \
-		sed 's/^/* Documentation\//' $(BUILDDIR)/spelling/output.txt; \
-		echo "If the words are not misspelled, run:"; \
-		echo -n "Documentation/update-spelling_wordlist.sh "; \
-		echo "`sed -E 's/^([^:]+:){2} \((.*)\)/\2/g' $(BUILDDIR)/spelling/output.txt | sort -u | tr -d '\r\n' | sed "s/'/\\\\\\\\'/g"`"; \
-		echo ""; \
-		exit 1; \
-	fi
-	#                                                                                                               ^^^^^^ replace FOO'bar with FOO\'bar
-	@rm $@.ok 2>/dev/null
+build-image:
+	$(QUIET)tar c requirements.txt Dockerfile \
+	  | docker build --tag cilium/docs-builder -
 
-# To use checklinks with `make render-docs`,
-# edit the line `%: Makefile check-requirements check-build checklinks`
-checklinks:
-	$(QUIET)$(SPHINXBUILD) -b linkcheck -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS)
+DOCKER_RUN := docker container run --rm \
+		--workdir /src/Documentation \
+		--volume $(CURDIR)/..:/src \
+			cilium/docs-builder
 
-.PHONY: clean
-# Removes build directory
-clean:
-	rm -rf $(BUILDDIR)
-.PHONY: help Makefile ../Makefile.quiet check-requirements cmdref
+cmdref:
+	@$(ECHO_GEN)cmdref
+	-$(QUIET)rm -rvf ${cmdref}/cilium*
+	$(QUIET)$(DOCKER_RUN) ./update-cmdrefs.sh
 
-# Catch-all target: route all unknown targets to Sphinx using the new
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile check-requirements check-build
+epub latex html: build-image
+	@$(ECHO_GEN)_api
+	$(QUIET)cp -r ../api _api
 	@$(ECHO_GEN)_build/$@
-	$(QUIET)$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+	$(QUIET)$(DOCKER_RUN) ./check-build.sh $(@)
+
+DOCS_PORT = 9081
+
+run-server: stop-server
+	$(QUIET)docker container run --rm \
```

Comment:
Can we make this run as a regular user UID/GID?

I ran the following commands and hit trouble:

```
$ make -C Documentation/
...
$ make -C Documentation/ clean
rm: cannot remove '_build/doctrees/environment.pickle': Permission denied
rm: cannot remove '_build/doctrees/gettingstarted/k8s-install-connectivity-test.doctree': Permission denied
rm: cannot remove '_build/doctrees/gettingstarted/kube-router.doctree': Permission denied
rm: cannot remove '_build/doctrees/gettingstarted/k8s-install-azure-cni-steps.doctree': Permission denied
...
$ ls -lh Documentation/_build/doctrees/environment.pickle
-rw-r--r-- 1 root root 704K Mar 13 10:17 Documentation/_build/doctrees/environment.pickle
```

Reply:
Good point, I'll see what can be done. I was only testing on Docker for Mac, and the [`osxfs`](https://docs.docker.com/docker-for-mac/osxfs/) does remaps UID/GID.

Reply:
See https://github.com/cilium/cilium/pull/10572/files#r392967432.